parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -148,7 +174,7 @@ public void writeTo(DataOutput out) throws IOException {
 
   private static class FromStringBinary extends ByteArrayBackedBinary {
     public FromStringBinary(byte[] value) {
-      super(value);
+      super(value, true);
```

Comment:
String to byte[] takes an immutable object and returns a mutable one. Then you create an unmodified variant of that mutable byte[]. It would be a better API to have the constructor here take a String rather than byte[]. That way, it cannot be passed a byte array that is referenced -- and possibly changed -- elsewhere.